packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -174,14 +191,16 @@ export async function uploadDialog(node: ZoweDatasetNode, datasetProvider: api.I
     }
 }
 
-export async function uploadFile(node: ZoweDatasetNode, doc: vscode.TextDocument) {
+export async function uploadFile(node: ZoweDatasetNode, doc: vscode.TextDocument): Promise<IZosFilesResponse> {
     try {
         const datasetName = node.label as string;
         const prof = node.getProfile();
-        await ZoweExplorerApiRegister.getMvsApi(prof).putContents(doc.fileName, datasetName, {
+
+        const response = await ZoweExplorerApiRegister.getMvsApi(prof).putContents(doc.fileName, datasetName, {
             encoding: prof.profile?.encoding,
             responseTimeout: prof.profile?.responseTimeout,
         });
+        return response;
     } catch (e) {
         await errorHandling(e, node.getProfileName(), e.message);
```

Comment:
It might have been the way I was testing, but I disconnected my internet in the middle of the upload and noticed that while there is an error dialog that comes back, there is also an error from `mDetails` being undefined in `errorHandling`:

![image](https://user-images.githubusercontent.com/86500639/227017552-33b37afb-633a-48bc-8405-421eb4bc21ed.png)

I think if we would add a `?` to line 50 of `ProfilesUtils.ts`, then that should resolve the issue:
```ts
if (errorDetails?.mDetails !== undefined)
```